modules/ting_search_carousel/ting_search_carousel.module

```diff
@@ -283,31 +283,30 @@ function _ting_search_carousel_find_entities_with_covers($query, $start, $size,
   $finished = FALSE;
   $entities = array();
 
-  $options = array(
-    'facets' => array(),
-    'enrich' => TRUE,
-    'allObjects' => FALSE,
-  );
+  $sal_query = ting_start_query()
+    ->setRawQuery($query)
+    ->setPage($start + 1)
+    ->setCount($size)
+    ->addFieldFilter('enrich', TRUE)
+    ->addFieldFilter('allObjects', FALSE)
+    ->setMaterialFilter($ignore);
 
-  // TODO: use the Search Abstraction Layer (ting_start_query()).
-  // Ting pages are 1, not 0, based.
-  $result = opensearch_do_search($query, ($start++) + 1, $size, $options);
+  $results = $sal_query->execute();
+  $collections = $results->getTingEntityCollections();
```

Comment:
Do we need to extract collections this early? As I see it we might as well put it in the `foreach` on L306:

`foreach ($results->getTingEntityCollections() as $collection) {`